src/application-base/ApplicationBase.jsx

```diff
@@ -40,36 +32,10 @@ const propTypes = {
    * If the `locale` prop is not provided, the preferred language from the browser will be used.
    */
   locale: PropTypes.string,
-  /**
-   * Custom translations for the current locale.
-   */
-  customTranslatedMessages: (props, propName, componentName) => {
-    if (!props[propName]) {
-      return null;
-    }
-
-    if (Object.keys(props[propName]).length !== 0 && props.locale === undefined) {
-      return new Error(`Missing locale prop for ${propName} in ${componentName} props`);
-    }
-
-    return null;
-  },
-  /**
-   * The component to render while the translation files are being retrieved.
-   * NOTE: Absolutely no locale-dependent logic should be
-   * utilized in this placeholder.
-   */
-  translationsLoadingPlaceholder: PropTypes.node,
   /**
    * The name of the theme to apply to the application using terra-theme-provider.
    */
   themeName: PropTypes.string,
```

Comment:
should this theme name prop should accept only the list of theme class name which terra support as of now like do we need to do that validation here ?. i know that this is not the correct pr for doing that. but just a thought.